api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/PaymentCycleScheduler.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.dhsc.htbhf.claimant.scheduler;
+
+import lombok.extern.slf4j.Slf4j;
+import net.javacrumbs.shedlock.core.SchedulerLock;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import uk.gov.dhsc.htbhf.claimant.repository.ClosingPaymentCycle;
+import uk.gov.dhsc.htbhf.claimant.repository.PaymentCycleRepository;
+
+import java.time.LocalDate;
+import java.util.List;
+
+@Component
+@Slf4j
+public class PaymentCycleScheduler {
+
+    private final PaymentCycleRepository paymentCycleRepository;
+    private final CreateNewPaymentCycleJob job;
+    private final int endDateOffsetDays;
+
+    public PaymentCycleScheduler(
+            PaymentCycleRepository paymentCycleRepository,
+            CreateNewPaymentCycleJob job,
+            @Value("${new-payment-cycle-schedule.end-date-offset-days:0}") int endDateOffsetDays) {
+
+        this.paymentCycleRepository = paymentCycleRepository;
+        this.job = job;
+        this.endDateOffsetDays = endDateOffsetDays;
+    }
+
+    @Scheduled(cron = "${new-payment-cycle-schedule.cron-schedule}")
+    @SchedulerLock(
+            name = "Create new payment cycles",
+            lockAtLeastForString = "${new-payment-cycles.minimum-lock-time}",
+            lockAtMostForString = "${new-payment-cycles.maximum-lock-time}")
+    public void createNewPaymentCycles() {
+        LocalDate cycleEndDate = LocalDate.now().plusDays(endDateOffsetDays);
+        log.debug("Querying for active claims with cycles ending on {}", cycleEndDate);
+        List<ClosingPaymentCycle> cycles = paymentCycleRepository.findActiveClaimsWithCycleEndingOnOrBefore(cycleEndDate);
+
```

Comment:
Is making a payment coming in the next PR?

Reply:
CreateNewPaymentCycleJob - creating a PaymentCycle and sending a MAKE_PAYMENT message (similar to your own work) comes next. Make payment comes after that.

Reply:
Correction CreateNewPaymentCycleJob will create a DETERMINE_ENTITLEMENT message, not, MAKE_PAYMENT